src/transformers/modeling_flax_utils.py

```diff
@@ -348,8 +349,19 @@ def from_pretrained(
             with open(resolved_archive_file, "rb") as state_f:
                 try:
                     state = from_bytes(cls, state_f.read())
-                except UnpicklingError:
-                    raise EnvironmentError(f"Unable to convert {archive_file} to Flax deserializable object. ")
+                except (UnpicklingError, msgpack.exceptions.ExtraData):
+                    try:
+                        with open(resolved_archive_file) as f:
+                            if f.read().startswith("version"):
+                                raise OSError(
+                                    "You seem to have cloned a repository without having git-lfs installed. Please install "
+                                    "git-lfs and run `git lfs install` followed by `git lfs pull` in the folder "
+                                    "you cloned."
+                                )
+                            else:
+                                raise ValueError
```

Comment:
No message in the error? Also it should be `from e` with `e` being the previous error caught by the first except. This makes it easier to see the original source of an error if someone has some try except around our code.

Reply:
I haven't put a message in the error because this `raise ValueError` is here simply to be caught by the try/assert around it:
```
except (UnicodeDecodeError, ValueError):
```

This ensures the exact same error as before is raised from a non-git-lfs weight file:

```
raise EnvironmentError(f"Unable to convert {archive_file} to Flax deserializable object. ")
```
See previous error:
```
                except UnpicklingError:
                    raise EnvironmentError(f"Unable to convert {archive_file} to Flax deserializable object. ")
```